packages/google_sign_in/google_sign_in_platform_interface/lib/google_sign_in_platform_interface.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:meta/meta.dart' show required, visibleForTesting;
+import 'src/method_channel_google_sign_in.dart';
+import 'src/types.dart';
+
+export 'src/method_channel_google_sign_in.dart';
+export 'src/types.dart';
+
+/// The interface that implementations of google_sign_in must implement.
+///
+/// Platform implementations that live in a separate package should extend this
+/// class rather than implement it as `google_sign_in` does not consider newly
+/// added methods to be breaking changes. Extending this class (using `extends`)
+/// ensures that the subclass will get the default implementation, while
+/// platform implementations that `implements` this interface will be broken by
+/// newly added [GoogleSignInPlatform] methods.
+abstract class GoogleSignInPlatform {
+  /// Only mock implementations should set this to `true`.
+  ///
+  /// Mockito mocks implement this class with `implements` which is forbidden
+  /// (see class docs). This property provides a backdoor for mocks to skip the
+  /// verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
+
+  /// The default instance of [GoogleSignInPlatform] to use.
+  ///
+  /// Platform-specific plugins should override this with their own
+  /// platform-specific class that extends [GoogleSignInPlatform] when they
+  /// register themselves.
+  ///
+  /// Defaults to [MethodChannelGoogleSignIn].
+  static GoogleSignInPlatform get instance => _instance;
+
+  static GoogleSignInPlatform _instance = MethodChannelGoogleSignIn();
+
+  // TODO(amirh): Extract common platform interface logic.
+  // https://github.com/flutter/flutter/issues/43368
+  static set instance(GoogleSignInPlatform instance) {
+    if (!instance.isMock) {
+      try {
+        instance._verifyProvidesDefaultImplementations();
+      } on NoSuchMethodError catch (_) {
+        throw AssertionError(
+            'Platform interfaces must not be implemented with `implements`');
+      }
+    }
+    _instance = instance;
+  }
+
+  /// This method ensures that [GoogleSignInPlatform] isn't implemented with `implements`.
+  ///
+  /// See class docs for more details on why using `implements` to implement
+  /// [GoogleSignInPlatform] is forbidden.
+  ///
+  /// This private method is called by the [instance] setter, which should fail
+  /// if the provided instance is a class implemented with `implements`.
+  void _verifyProvidesDefaultImplementations() {}
+
+  /// Initializes the plugin. You must call this method before calling other methods.
+  /// See: https://developers.google.com/identity/sign-in/web/reference#gapiauth2initparams
+  Future<void> init(
+      {@required String hostedDomain,
+      List<String> scopes,
+      SignInOption signInOption,
+      String clientId}) async {
+    throw UnimplementedError('init() has not been implemented.');
+  }
+
+  /// Attempts to reuse pre-existing credentials to sign in again, without user interaction
```

Comment:
Nit: add a period at the end of the sentence

Reply:
Yep, good catch! (I hoped this was a checkstyle type lint!)